packages/image_picker/image_picker_for_web/lib/image_resizer_utils.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+import 'dart:ui';
+
+///a function that checks if an image needs to be resized or not
+bool imageResizeNeeded(double? maxWidth, double? maxHeight, int? imageQuality) {
+  bool resizeNeeded =
+      (maxWidth != null || maxHeight != null || imageQuality != null);
+  if (resizeNeeded) {
+    if (imageQuality != null) {
+      return isImageQualityValid(imageQuality);
+    } else {
+      return true;
+    }
+  } else {
+    return false;
+  }
+}
+
+/// a function that checks if image quality is between 0 to 100
+bool isImageQualityValid(int imageQuality) {
+  return (imageQuality >= 0 && imageQuality <= 100);
+}
+
+/// a functions that calculates the size of the scaled image.
+/// imageWidth is the width of the image
+/// imageHeight is the height of  the image
+/// maxWidth is the maximum width of the scaled image
+/// maxHeight is the maximum height of the scaled image
+Size calculateSizeOfScaledImage(double imageWidth, double imageHeight,
+    double? maxWidth, double? maxHeight) {
+  double originalWidth = imageWidth;
+  double originalHeight = imageHeight;
+
+  bool hasMaxWidth = maxWidth != null;
```

Comment:
If we're not stretching the output of the image, I think most of this method can be simplified a little bit:

```dart
Size calculateSizeOfScaledImage(Size imageSize, double? maxWidth, double? maxHeight) {
  double widthFactor = maxWidth != null ? imageSize.width / maxWidth : 0;
  double heightFactor = maxHeight != null ? imageSize.height / maxHeight : 0;
  double resizeFactor = Math.max(widthFactor, heightFactor);
  return imageSize ~/ resizeFactor;
}
```

Reply:
This does work but it scales the image above its original width and height . I think I should have used the word downScaled instead of scaled in this function .  The android and IOS implementation of this function always downScales the  image and never upscales the image.  so I think, I will keep the same implementation as the others to be consistent in all platforms . What do you think?

Reply:
> The android and IOS implementation of this function always downScales the image and never upscales the image.

Absolutely, yes!

Reply:
> it scales the image above its original width and height

only if `resizeFactor` is < 1, I think the upscaling can be avoided by:

```
return (resizeFactor > 1 ? imageSize ~/ resizeFactor : imageSize);
```

Right? No need for the super complex algorithm to compute the resized dimensions?

Reply:
yes